Runtimes/Supplemental/cmake/modules/FindSwiftOverlay.cmake

```diff
@@ -0,0 +1,236 @@
+
+ ``swiftDarwin_DIR_FOUND``
+   true if the overlay was found
+
+ ``swiftDarwin_INCLUDE_DIR``
+   the directory containing the corresponding overlay swift module folder
+
+The module may set the following variables if `swiftGlibc_DIR` is not set and targeting the GlibC overlay.
+
+ ``swiftGlibc_DIR_FOUND``
+   true if the overlay was found
+
+ ``swiftGlibc_INCLUDE_DIR``
+   the directory containing the corresponding overlay swift module folder
+
+The module may set the following variables if `swiftWinSDK_DIR` is not set and targeting the WinSDK overlay.
+
+ ``swiftWinSDK_DIR_FOUND``
+   true if the overlay was found
+
+ ``swiftWinSDK_INCLUDE_DIR``
+   the directory containing the corresponding overlay swift module folder
+
+The module may set the following variables if `swiftCRT_DIR` is not set and targeting the CRT overlay.
+
+ ``swiftCRT_DIR_FOUND``
+   true if the overlay was found
+
+ ``swiftCRT_INCLUDE_DIR``
+   the directory containing the corresponding overlay swift module folder
+
+The module may set the following variables if `swiftAndroid_DIR` is not set and targeting the Android overlay.
+
+ ``swiftAndroid_DIR_FOUND``
+   true if the overlay was found
+
+ ``swiftAndroid_INCLUDE_DIR``
+   the directory containing the corresponding overlay swift module folder
+
+
+#]=======================================================================]
+
+include_guard(GLOBAL)
+
+include(FindPackageHandleStandardArgs)
+include(PlatformInfo)
+
+# This was loosely modelled after other find modules
+# (namely FindGLEW), where the equivalent parameter
+# is not stored in cache (possibly because we want
+# the project importing it to be able to
+# it "immediately")
+if(NOT DEFINED SwiftOverlay_USE_STATIC_LIBS)
+  set(SwiftOverlay_USE_STATIC_LIBS OFF)
+  if(NOT BUILD_SHARED_LIBS AND NOT APPLE)
+    set(SwiftOverlay_USE_STATIC_LIBS ON)
+  endif()
+endif()
+
+if(SwiftOverlay_DIR)
+  find_package(SwiftOverlay CONFIG)
+  return()
+endif()
+
+if(APPLE)
+  list(APPEND OVERLAY_TARGET_NAMES "swiftDarwin")
+  # Look in the SDK
+  list(APPEND swiftDarwin_INCLUDE_DIR_HINTS
+    "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
+  list(APPEND swiftDarwin_LIBRARY_HINTS
+    "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
+  # When building for Apple platforms, swiftDarwin always comes from within the
+  # SDK as a tbd for a shared library in the shared cache.
+  list(APPEND swiftDarwin_NAMES libswiftDarwin.tbd)
+  set(swiftDarwin_MODULE_NAME "Darwin.swiftmodule")
+elseif(LINUX)
+  #ToDo(swiftlang/swift/issues/83014): Handle the static MUSL SDK case
+  list(APPEND OVERLAY_TARGET_NAMES "swiftGlibc")
+
+  # If the swiftGlibc_DIR_FLAG is specified, look there instead. The cmake-generated
+  # config file is more accurate
+  if(swiftGlibc_DIR)
+    if(swiftGlibc_FIND_REQUIRED)
+      list(APPEND args REQUIRED)
+    endif()
+    if(swiftGlibc_FIND_QUIETLY)
+      list(APPEND args QUIET)
+    endif()
+    find_package(swiftGlibc NO_MODULE ${args})
+  # Look in the SDK
+  else()
```

Comment:
I don't think this is needed -- I expect `swiftGlibc` to be exported as part of the `Overlay` build system, so the call to ` find_package(SwiftOverlay ...)` should take care of this.

```suggestion
```

(the matching `endif()` should be delete as well)